lib/translator/inquiry.cc

```diff
@@ -342,17 +342,19 @@ StatusCode InquiryToScsi(Span<const uint8_t> raw_scsi, Span<uint8_t> buffer,
   uint8_t* ns_dptr = reinterpret_cast<uint8_t*>(nvme_cmds[1].dptr.prp.prp1);
   Span<uint8_t> ns_span(ns_dptr, sizeof(nvme::IdentifyNamespace));
 
-  nvme::IdentifyControllerData identify_ctrl = {};
-  if (!ReadValue(ctrl_span, identify_ctrl)) {
-    DebugLog("Malformed IdentifyController Data");
-    return StatusCode::kInvalidInput;
-  };
+  const nvme::IdentifyControllerData* identify_ctrl =
+      SafePointerCastRead<nvme::IdentifyControllerData>(ctrl_span);
+  if (identify_ctrl == nullptr) {
+    DebugLog("Identify controller structure failed to cast");
+    return StatusCode::kFailure;
+  }
 
-  nvme::IdentifyNamespace identify_ns = {};
-  if (!ReadValue(ns_span, identify_ns)) {
-    DebugLog("Malformed IdentifyNamespace Data");
-    return StatusCode::kInvalidInput;
-  };
+  const nvme::IdentifyNamespace* identify_ns =
+      SafePointerCastRead<nvme::IdentifyNamespace>(ns_span);
+  if (identify_ns == nullptr) {
+    DebugLog("Identify namespace structure failed to cast");
+    return StatusCode::kFailure;
```

Comment:
I noticed that we don't have a test for this case. Do you think it's worth adding one?

Reply:
Good point, I went ahead & added some basic tests to ensure the nullptr case is caught by the functions